core/kernel/interrupt.c

```diff
@@ -62,79 +90,223 @@ void itr_handle(size_t it)
 	}
 
 	if (!was_handled) {
-		EMSG("Disabling unhandled interrupt %zu", it);
-		itr_chip->ops->disable(itr_chip, it);
+		EMSG("Mask unhandled interrupt %s:%zu", chip->name, itr_num);
+		interrupt_mask(chip, itr_num);
 	}
 }
 
-struct itr_handler *itr_alloc_add_type_prio(size_t it, itr_handler_t handler,
-					    uint32_t flags, void *data,
-					    uint32_t type, uint32_t prio)
+TEE_Result interrupt_configure(struct itr_chip *chip, size_t itr_num,
+			       uint32_t type, uint32_t prio)
 {
-	struct itr_handler *hdl = calloc(1, sizeof(*hdl));
+	chip->ops->add(chip, itr_num, type, prio);
 
-	if (hdl) {
-		hdl->it = it;
-		hdl->handler = handler;
-		hdl->flags = flags;
-		hdl->data = data;
-		itr_add_type_prio(hdl, type, prio);
+	return TEE_SUCCESS;
+}
+
+TEE_Result interrupt_add_configure_handler(struct itr_handler *hdl,
+					   uint32_t type, uint32_t prio)
+{
+	struct itr_handler *h = NULL;
+
+	assert(hdl && hdl->chip->ops);
+
+	SLIST_FOREACH(h, &hdl->chip->handlers, link) {
```

Comment:
Should it be a SAFE loop has the remove will update the list?

Reply:
This part was merged with https://github.com/OP-TEE/optee_os/pull/6096.
That said, I agree, theree should be some protection here in case interrupt handlers are added/removed at runtime.